docs/upgrade.txt

```diff
@@ -11,39 +11,92 @@ Upgrading
-  ``Query\Builder::where($column, 'all', $values)``
+.. _laravel-breaking-changes:
 
-- ``Query\Builder::delete()`` doesn't accept ``limit()`` other than ``1`` or ``null``.
+Breaking Changes
+----------------
 
-- ``whereDate``, ``whereDay``, ``whereMonth``, ``whereYear``, ``whereTime``
-  now use MongoDB operators on date fields
+A breaking change is a modification in a convention or behavior in
+a specific version of {+odm-short+} that may prevent your application from
+working as expected.
 
-- Replace ``Illuminate\Database\Eloquent\MassPrunable`` with ``MongoDB\Laravel\Eloquent\MassPrunable``
-  in your models
+The breaking changes in this section are categorized by the major
+version releases that introduced them. When upgrading ODM versions,
+address all the breaking changes between your current version and the
+planned upgrade version.
 
-- Remove calls to not-supported methods of ``Query\Builder``: ``toSql``,
+.. _laravel-breaking-changes-v4.x:
+
+Version 4.x Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This ODM version introduces the following breaking changes:
+
+- Minimum Laravel version is now 10.0. For instructions on upgrading your Laravel version,
+  see the `Upgrade Guide <https://laravel.com/docs/10.x/upgrade>`__ in the Laravel
+  documentation.
+
+- Dependency name is now ``"mongodb/laravel-mongodb"``. Ensure that the dependency
+  name in your ``composer.json`` file is ``"mongodb/laravel-mongodb": "^4.0"``. Then, run
+  ``composer update``.
+
+- Namespace is now ``MongoDB\Laravel\``. Ensure that you change the namespace from ``Jenssegers\Mongodb\``
+  to ``MongoDB\Laravel\`` in your models and config files.
+
+- Removes support for non-Laravel projects.
+
+- Removes support for the ``$dates`` property. Ensure that you change all instances of ``$dates``
+  to ``$casts`` in your model files.
+
+- ``Model::unset($field)`` does not persist the change. Ensure that you follow all calls to
+  ``Model::unset($field)`` with ``Model::save()``.
+
+- Removes the ``Query\Builder::whereAll($column, $values)`` method. Ensure that you replace all calls
+  to ``Query\Builder::whereAll($column, $values)`` with ``Query\Builder::where($column, 'all', $values)``.
+
+- ``Query\Builder::delete()`` can only delete one or all documents. Ensure that you only pass a value 
+  of ``1`` or ``null`` to ``limit()``.
+
+- ``whereDate()``, ``whereDay()``, ``whereMonth()``, ``whereYear()``, and ``whereTime()`` methods
+  now use MongoDB operators on date fields.
+
+- Adds the ``MongoDB\Laravel\Eloquent\MassPrunable`` trait. Ensure that you replace all instances of 
+  ``Illuminate\Database\Eloquent\MassPrunable`` with ``MongoDB\Laravel\Eloquent\MassPrunable``
+  in your models.
+
+- Removes support for the ``Query\Builder``: ``toSql``,
   ``toRawSql``, ``whereColumn``, ``whereFullText``, ``groupByRaw``,
   ``orderByRaw``, ``unionAll``, ``union``, ``having``, ``havingRaw``,
   ``havingBetween``, ``whereIntegerInRaw``, ``orWhereIntegerInRaw``,
```

Comment:
I think this might work better as a list so it's not such a big block of methods:

```suggestion
Removes support for the following ``Query\Builder`` methods:

 - ``toSql``
 ...
 ...